src/common/facetUtils.js

```diff
@@ -0,0 +1,56 @@
+export function toggleViewLess(stateKey, event) {
+    const facetName = event.target.dataset['unx_name'];
+    this.setState((existingState) => {
+        const updatedFacets = existingState[stateKey].map((facetValue) => {
+            if (facetValue.facetName === facetName) {
+                return {
+                    ...facetValue,
+                    viewLess: !facetValue.viewLess
+                };
+            }
+            return { ...facetValue };
```

Comment:
if `facetValue` is an object, we can just do `return facetvalue` ?

Reply:
Shouldn't we be passing new data. functional programming and all? So react detects a change.